DEV-3227: Reformat Example (#106)
cloudposse-examples/atmos-pro-example-advanced: .github/README.md

# atmos-pro-example-advanced

An advanced example demonstrating how to use Atmos Pro with GitHub Actions for infrastructure deployment. This
repository shows a comprehensive setup with multiple components and dependencies, showcasing how Atmos Pro handles
complex infrastructure workflows.

## Introduction

This is an advanced example repository that demonstrates how to use Atmos Pro with GitHub Actions for infrastructure
deployment. It provides a comprehensive setup with multiple components and dependencies to showcase how Atmos Pro
handles complex infrastructure workflows.

For comprehensive documentation and advanced features, visit [atmos-pro.com/docs](https://atmos-pro.com/docs).

## Usage

### Installation

<details>

> <summary><strong>Prerequisites for Using Atmos CLI</strong></summary>

Reviewer: ```suggestion
<summary><strong>Prerequisites for Running Atmos Locally</strong></summary>
```

**Option A: Use our reference architecture**

- For Terraform state backend setup:
  [Initialize Terraform State Backend](https://docs.cloudposse.com/layers/accounts/initialize-tfstate/)
- For plan storage setup: [Setup GitOps with GitHub Actions](https://docs.cloudposse.com/layers/gitops/setup/)
- For GitHub OIDC setup:
  [How to use GitHub OIDC with AWS](https://docs.cloudposse.com/layers/github-actions/github-oidc-with-aws/)

**Option B: Use the CloudFormation template**

- Deploy all three components at once using the
  [CloudFormation template for Terraform backend](https://github.com/cloudposse/aws-cloudformation-terraform-backend)
- This template creates the S3 buckets, DynamoDB tables, and GitHub OIDC provider in a single deployment

</details>

### Quick Start

Follow these steps to get started with this Atmos Pro example:

1. **Fork this repository**

   - Click the "Fork" button in the top right corner of this repository on GitHub

2. **Deploy prerequisites**

   - See the "Required Prerequisites for Using Atmos Pro (Advanced Example)" section above for deployment options

3. **Install the Atmos Pro GitHub App**

>    - Follow the [Atmos Pro documentation](https://atmos-pro.com/docs) to sign up for Atmos Pro and install the GitHub

Reviewer: ```suggestion
   - Follow the [Atmos Pro documentation](https://atmos-pro.com/docs/install) to sign up for Atmos Pro and install the GitHub
```

- **Terraform or OpenTofu** - Atmos requires either Terraform or OpenTofu to be installed locally. For installation
  instructions:
  - Terraform: [Install Terraform](https://developer.hashicorp.com/terraform/downloads)
  - OpenTofu: [Install OpenTofu](https://opentofu.org/docs/intro/install/)

</details>

<details>
<summary><strong>Required Prerequisites for Using Atmos Pro (Advanced Example)</strong></summary>

The following prerequisites are **required** to use this advanced example with real infrastructure:

#### Required Infrastructure Components

This advanced example requires three infrastructure components to be deployed:

1. **Terraform State Backend** - Remote state storage (S3 bucket and DynamoDB table for state locking)
2. **Terraform Plan Storage Backend** - Plan file storage (S3 bucket and DynamoDB table for storing Terraform plans
   between planning and applying phases)
3. **GitHub OIDC Integration** - Secure authentication (GitHub OIDC provider and IAM roles for GitHub Actions to
   authenticate with AWS)

#### Deployment Options

You can deploy these infrastructure components using one of the following approaches:

> **Option A: Use our reference architecture**

Reviewer: ```suggestion
**Option A: Use Terraform Components**
```

> - For Terraform state backend setup:

Reviewer: ```suggestion
- For the Terraform state backend setup, see:
```

  [Initialize Terraform State Backend](https://docs.cloudposse.com/layers/accounts/initialize-tfstate/)

> - For plan storage setup: [Setup GitOps with GitHub Actions](https://docs.cloudposse.com/layers/gitops/setup/)

Reviewer: ```suggestion
- For the plan storage setup. see: [Setup GitOps with GitHub Actions](https://docs.cloudposse.com/layers/gitops/setup/)
```

> - For GitHub OIDC setup:

Reviewer: ```suggestion
- For the GitHub OIDC setup, see:
```

  [How to use GitHub OIDC with AWS](https://docs.cloudposse.com/layers/github-actions/github-oidc-with-aws/)

**Option B: Use the CloudFormation template**

> - Deploy all three components at once using the

Reviewer: ```suggestion
- Deploy all three dependencies at once using the
```

> # atmos-pro-example-advanced
> 
> An advanced example demonstrating how to use Atmos Pro with GitHub Actions for infrastructure deployment. This
> repository shows a comprehensive setup with multiple components and dependencies, showcasing how Atmos Pro handles
> complex infrastructure workflows.
> 
> ## Introduction
> 
> This is an advanced example repository that demonstrates how to use Atmos Pro with GitHub Actions for infrastructure
> deployment. It provides a comprehensive setup with multiple components and dependencies to showcase how Atmos Pro
> handles complex infrastructure workflows.
> 
> For comprehensive documentation and advanced features, visit [atmos-pro.com/docs](https://atmos-pro.com/docs).

Reviewer: ```suggestion
# atmos-pro-example-advanced

An advanced reference repository showing how to use Atmos Pro with GitHub Actions to deploy infrastructure.

## Introduction

This example goes beyond the basics to demonstrate a full-featured setup, including multiple components, dependencies, and coordinated workflows.  
It’s designed to help you understand how Atmos Pro streamlines complex infrastructure management in real-world scenarios.

For comprehensive documentation and advanced features, visit [atmos-pro.com/docs](https://atmos-pro.com/docs).
```
